providers/curation/github.js

```diff
@@ -144,15 +145,60 @@ class GitHubCurationService {
     return serviceGithub.repos.createFile(fileBody)
   }
 
+  // Return an array of valid patches that exist
+  // and a list of definitions that do not exist in the store
+  async _validateComponentDefinitionExists(patches) {
+    const coordinatesWithRevisions = flatten(
+      map(patches, patch =>
+        map(patch.revisions, (_, key) => {
+          return Object.assign({}, patch.coordinates, {
+            revision: key
+          })
+        })
+      )
+    )
+
+    const validDefinitions = await this.definitionService.listAll(coordinatesWithRevisions)
+    const validPatches = Object.assign([], patches)
+    const notFoundDefinitions = []
+    map(patches, (patch, index) =>
+      map(
+        patch.revisions,
+        throat(10, async (_, key) => {
+          const coordinates = Object.assign({}, patch.coordinates, {
+            revision: key
+          })
+          if (validDefinitions.findIndex(definition => isEqual(definition, coordinates)) === -1) {
+            delete validPatches[index].revisions[key]
+            if (isEmpty(validPatches[index].revisions)) delete validPatches[index]
+            notFoundDefinitions.push(
+              `Definition ${EntityCoordinates.fromObject(coordinates).toString()} has not been found`
+            )
+          }
+        })
+      )
+    )
+    return { validPatches: compact(validPatches), notFoundDefinitions }
+  }
+
   async addOrUpdate(userGithub, serviceGithub, info, patch) {
     const { owner, repo, branch } = this.options
     const masterBranch = await serviceGithub.repos.getBranch({ owner, repo, branch: `refs/heads/${branch}` })
     const sha = masterBranch.data.commit.sha
     const prBranch = await this._getBranchName(info)
     await serviceGithub.gitdata.createReference({ owner, repo, ref: `refs/heads/${prBranch}`, sha })
+    const { validPatches, notFoundDefinitions } = await this._validateComponentDefinitionExists(patch.patches)
+    const validResult = { errors: notFoundDefinitions }
+    if (validPatches.length === 0) {
+      validResult.errors.push(
+        'The contribution has failed because none of the supplied component definitions were found'
+      )
+      return validResult
+    }
     await Promise.all(
-      patch.patches.map(throat(1, component => this._writePatch(serviceGithub, info, component, prBranch)))
+      validPatches.map(throat(10, validPatch => this._writePatch(serviceGithub, info, validPatch, prBranch)))
```

Comment:
Believe we actually have to use 1 as GitHub balks at multiple concurrent patches being written on the same branch or some such.

Reply:
@jeffmcaffer We set that to 10 as you originally suggested to use that setup. @moranthomas can you restore it to 1?

Reply:
That's an easy rollback. No problem

Reply:
When you roll this back, perhaps add a code comment to the effect that it MUST be 1 or GitHub gets upset.

Reply:
I've added a comment on this instruction